### PR TITLE
Enforce domain verfication if supplied in config

### PR DESCRIPTION
**Pull request by satterly**

### CODE REVIEW

File: module/src/main/scala/com/gu/googleauth/auth.scala [hunk 1]
```diff
@@ -108,6 +108,10 @@ object GoogleAuth {
           googleResponse(response) { json =>
               val token = Token.fromJson(json)
               val jwt = token.jwt
+              config.domain foreach { domain =>
+                if (!jwt.claims.email.split("@").lastOption.contains(domain))
```
Comment (rrees) [inline module/src/main/scala/com/gu/googleauth/auth.scala]: contains or exactly matches?

Comment (philwills) [inline module/src/main/scala/com/gu/googleauth/auth.scala]: :+1:  for an exact match.

Also UserIdentity is already extracting the domain, see https://github.com/guardian/play-googleauth/blob/d4b6085203c75ce8576cce91d1cfc8a4453bc92f/module/src/main/scala/com/gu/googleauth/actions.scala#L11, so in the mid-term, I'd suggest we use that, but I'm happy to merge this now and tidy later.

**Comment by sihil** [comment]
I think we should validate the hd value matches in the claim token we get back. Although it could always return the correct but unenforced value which wouldn't be ideal.

File: module/src/main/scala/com/gu/googleauth/auth.scala (hunk 1 above)
Comment (adamnfish) [inline module/src/main/scala/com/gu/googleauth/auth.scala]: Contains is an exact match - it's an option and you're checking the Option contains that value

**Comment by adamnfish** [comment]
I think this is ok for now to close the hole, as @philwills and @sihil have said, there are a couple of changes to make afterwards. I propose we properly implement the recommended ID token checks in another PR once this fix is out. Here's the list of Google's recommendations:

https://developers.google.com/identity/protocols/OpenIDConnect?hl=en#hd-param under `Validation of an ID token requires several steps`.

**Comment by adamnfish** [comment]
:+1: for this though

**Comment by sihil** [comment]
released as 0.3.3 to sonatype.

**Pull request merged**